app/frontend/src/features/BugReport.tsx

```diff
@@ -0,0 +1,173 @@
+import { darken, makeStyles, useMediaQuery, useTheme } from "@material-ui/core";
+import { Error as GrpcError } from "grpc-web";
+import React, { useState } from "react";
+import { useForm } from "react-hook-form";
+import { useMutation } from "react-query";
+
+import Alert from "../components/Alert";
+import Button from "../components/Button";
+import {
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+} from "../components/Dialog";
+import { BugIcon } from "../components/Icons";
+import TextField from "../components/TextField";
+import { service } from "../service";
+import { useAuthContext } from "./auth/AuthProvider";
+
+export interface BugReportFormData {
+  subject: string;
+  description: string;
+  steps: string;
+  results: string;
+}
+
+const useStyles = makeStyles((theme) => ({
+  button: {
+    backgroundColor: theme.palette.error.main,
+    "&:hover": {
+      backgroundColor: darken(theme.palette.error.main, 0.1),
+    },
+  },
+  field: {
+    "& + &": {
+      marginBlockStart: theme.spacing(2),
+    },
+  },
+  startIcon: {
+    [theme.breakpoints.down("sm")]: {
+      margin: 0,
+    },
+  },
+}));
+
+export default function BugReport() {
+  const theme = useTheme();
+  const isBelowMd = useMediaQuery(theme.breakpoints.down("sm"));
+
+  const classes = useStyles();
+  const [isOpen, setIsOpen] = useState(false);
+  const {
+    register,
+    handleSubmit,
+    reset: resetForm,
+  } = useForm<BugReportFormData>();
+  const userId = useAuthContext().authState.userId;
+  const {
+    data: reportIdentifier,
+    error,
+    isLoading,
+    mutate: reportBug,
```

Comment:
there is also a `mutateAsync` function if you want a promise (I think you mentioned that?)

Reply:
But then we'd have to handle errors in a try-catch. I prefer the non-async version (but still async under the hood) but with all the logic done in the `on*` callbacks to keep everything together in the `useMutation` so everything's nicely colocated, so I think I'll leave this as it is.